### PR TITLE
TextEditor: hacks to re-enable rendering on Firefox and copy/paste

### PR DESCRIPTION
**Pull request by maximlt**
Fixes https://github.com/holoviz/panel/issues/5545
Fixes https://github.com/holoviz/panel/issues/5600

"Fixes" is a bit exaggerated!

Basically I've noticed that the selection parsed by `normalizeNative` contained a node the code didn't expect so I tried to return `null` in that case and it worked (even if there's still an error in the console...).

On rendering in Firefox, browsers have different level of support for the Selection API and patching it for Firefox made things worse obviously, so I've disabled that and the widget can be rendered again and interacted with (there are still some weird behaviors).

Until Quill supports being rendered in the Shadow DOM, or we find a proper way to patch it, the TextEditor widget is going to behave strangely in some cases.

### CODE REVIEW

File: panel/models/quill.ts [hunk 1]
```diff
@@ -101,13 +106,17 @@ export class QuillInputView extends HTMLBoxView {
...
 
-    // Hack Quill and replace document.getSelection with shadow.getSelection
-    // see https://stackoverflow.com/questions/67914657/quill-editor-inside-shadow-dom/67944380#67944380
-    this.quill.selection.getNativeRange = () => {
-      const selection = (this.shadow_el as any).getSelection();
-      const range = normalizeNative(selection);
-      return range;
-    };
+    // Apply only with getSelection() is defined (e.g. undefined on Firefox)
+    if (typeof this.quill.root.getRootNode().getSelection !== 'undefined') {
```
Comment (Hoxbro) [inline panel/models/quill.ts:110]: Shouldn't this be inside the function? Or is `getSelection` never defined in FireFox?

Comment (maximlt) [reply panel/models/quill.ts:110]: Not sure about the correct language to use there, `getSelection` isn't "supported" in the shadow DOM on Firefox, see https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot#instance_methods.

I think that then when `document.getSelection()` is called on Firefox, it "pierces" through the shadow DOM. Of course, I have no idea what I'm talking about.

Comment (philippjfr) [reply panel/models/quill.ts:110]: >I have no idea what I'm talking about.

Fake til you make it 😄 That certainly sounded convincing.

**Review by philippjfr — APPROVED** [review]
Until Quill itself fixes thing this seems like hacky but reasonable fix.

**Pull request merged**